demo/pthread_demo.c

```diff
@@ -170,8 +171,8 @@ int server(const char *url)
 
     /*  Now wait on them to finish. */
     for (i = 0; i < MAXWORKERS; i++) {
-        if (tids[i] != NULL) {
-            pthread_join (tids[i], NULL);
+        if (tids[i] != '\0') {
```

Comment:
This is wrong.  NULL is the correct value here -- i.e. its a (void *)0   Not a character.

Reply:
As far as the other change (the htonl thing) -- maybe it would be better to simply eliminate the use of the macro altogether for this example.  Real programs should use endian safe operations if they want to be portable, but this example is about showing the use of threads, not about showing how to be portable.